parsers/src/main/java/org/iqu/parsers/HtmlEventsParser.java

```diff
@@ -0,0 +1,118 @@
+import java.text.SimpleDateFormat;
+import java.util.ArrayList;
+import java.util.List;
+
+import org.apache.log4j.Logger;
+import org.iqu.parsers.entities.Event;
+import org.jsoup.Jsoup;
+import org.jsoup.nodes.Document;
+import org.jsoup.nodes.Element;
+import org.jsoup.select.Elements;
+
+/**
+ * Class that implements a parser that makes an entity from html tags.
+ * 
+ * @author Razvan Rosu
+ *
+ */
+public class HtmlEventsParser implements Parser<Event> {
+
+	private Event event;
+	private static final Logger lOGGER = Logger.getLogger(HtmlEventsParser.class);
+	private static String pattern = "yyyy-MM-dd'T'HH:mm";
+	private static DateFormat dF = new SimpleDateFormat(pattern);
+	private long endDate = 0;
+	private String categories = "";
+	private String source;
+	private String dateEnd;
+
+	@Override
+	public List<Event> readFeed(String sourceURL, String encoding) {
+
+		List<Event> events = new ArrayList<Event>();
+		source = sourceURL;
+		Document doc = null;
+		try {
+			doc = Jsoup.connect(sourceURL).get();
+		} catch (IOException e) {
+			lOGGER.error("Error loading URL");
+		}
+		Elements item = doc.getElementsByClass(EVENTGROUP);
+
+		for (Element element : item) {
+
+			String UrlToConnect = element.getElementsByTag(A).attr(URL);
+			if (UrlToConnect != null && UrlToConnect.length() != 0)
+				takeEvents(element.getElementsByTag(A).attr(URL), events);
+		}
+
+		return events;
+	}
+
+	private void takeEvents(String eventsURL, List<Event> events) {
+		Document doc = null;
+		try {
+			doc = Jsoup.connect(eventsURL).get();
+		} catch (MalformedURLException e) {
+			lOGGER.error("Invalid URL!", e);
+		} catch (IOException e) {
+			lOGGER.error("InputStream error!", e);
+		}
+		categories = doc.body().id();
+		Elements item = doc.getElementsByClass(EVENTLIST);
+		for (Element element : item) {
+			event = new Event();
+			try {
+				getTags(element);
+				events.add(event);
+			} catch (ParseException e) {
+				lOGGER.error("Invalid tags");
+			}
+		}
+
+	}
+
+	private void getTags(Element element) throws ParseException {
+		event.setTitle(element.getElementsByTag(A).attr(TITLE));
+		event.setImage_id(element.getElementsByTag(IMG).attr(IMGSRC));
+		event.setExternal_url(element.getElementsByTag("a").attr(URL));
+		event.setDescription(element.getElementsByClass(DESCRIPTION).html());
+		event.setId(element.getElementsByClass(EVENTLIST).attr(ID) + "-" + categories);
+		event.setStartDate(convertDate(element.getElementsByTag(TIME).attr(DATE)));
+		event.setSource(source);
+		event.setCategories(categories);
+		dateEnd = element.getElementsByAttributeValue("itemprop", "endDate").attr(HtmlParserConstants.DATE);
+		if (dateEnd.length() > 0) {
+			try {
+				endDate = convertDate(dateEnd);
+			} catch (ParseException e) {
+				lOGGER.error("Invalid date format");
+			}
+		} else
+			endDate = 0;
+		event.setEndDate(endDate);
+	}
+
+	private long convertDate(String date) throws ParseException {
+		date = date.substring(0, 16);
```

Comment:
Replace this with the usage of a regex. e.g: http://stackoverflow.com/questions/18591242/java-extract-date-from-string-using-regex-failing